src/engine/sync/abletonlink.h

```diff
@@ -0,0 +1,137 @@
+using MixxxClockRef = ableton::platforms::stl::Clock;
+
+class AbletonLink : public QObject, public Syncable {
+    Q_OBJECT
+  public:
+    AbletonLink(const QString& group, EngineSync* pEngineSync);
+    ~AbletonLink() override = default;
+
+    const QString& getGroup() const override {
+        return m_group;
+    }
+    EngineChannel* getChannel() const override {
+        return nullptr;
+    }
+
+    /// Notify a Syncable that their mode has changed. The Syncable must record
+    /// this mode and return the latest mode in response to getMode().
+    void setSyncMode(SyncMode mode) override;
+
+    /// Notify a Syncable that it is now the only currently-playing syncable.
+    void notifyUniquePlaying() override;
+
+    /// Notify a Syncable that they should sync phase.
+    void requestSync() override;
+
+    /// Must NEVER return a mode that was not set directly via
+    /// notifySyncModeChanged.
+    SyncMode getSyncMode() const override;
+
+    /// Only relevant for player Syncables.
+    bool isPlaying() const override;
+    bool isAudible() const override;
+    bool isQuantized() const override;
+
+    /// Gets the current speed of the syncable in bpm (bpm * rate slider), doesn't
+    /// include scratch or FF/REW values.
+    mixxx::Bpm getBpm() const override;
+
+    /// Gets the beat distance as a fraction from 0 to 1
+    double getBeatDistance() const override;
+
+    /// Gets the speed of the syncable if it was playing at 1.0 rate.
+    mixxx::Bpm getBaseBpm() const override;
+
+    /// The following functions are used to tell syncables about the state of the
+    /// current Sync Master.
+    /// Must never result in a call to
+    /// SyncableListener::notifyBeatDistanceChanged or signal loops could occur.
+    void updateLeaderBeatDistance(double beatDistance) override;
+
+    /// Enforces the immediate change of the beat distance of all Link peers
+    void forceUpdateLeaderBeatDistance(double beatDistance);
+
+    /// Must never result in a call to SyncableListener::notifyBpmChanged or
+    /// signal loops could occur.
+    void updateLeaderBpm(mixxx::Bpm bpm) override;
+
+    void notifyLeaderParamSource() override;
+
+    /// Combines the above three calls into one, since they are often set
+    /// simultaneously.  Avoids redundant recalculation that would occur by
+    /// using the three calls separately.
+    void reinitLeaderParams(double beatDistance, mixxx::Bpm baseBpm, mixxx::Bpm bpm) override;
+
+    /// Must never result in a call to
+    /// SyncableListener::notifyInstantaneousBpmChanged or signal loops could
+    /// occur.
+    void updateInstantaneousBpm(mixxx::Bpm bpm) override;
+
+    void onCallbackStart(int sampleRate,
+            int bufferSize,
+            std::chrono::microseconds absTimeWhenPrevOutputBufferReachesDac);
+    void onCallbackEnd(int sampleRate, int bufferSize);
+
+  private:
+    ableton::link::HostTimeFilter<MixxxClockRef> m_hostTimeFilter;
+    const QString m_group;
```

Comment:
const member variables inhibit the possibility to move.
```suggestion
    QString m_group;
```

Reply:
But this must not move, it should stay constant for the lifetime of the class instance.

Reply:
Sure it can stay constant, but thats an invariant the class itself needs to hold up. If you mark the member constant, the entire class instance can't move anymore. Here's some more background on const member variables: https://www.sandordargo.com/blog/2020/11/11/when-use-const-2-member-variables

Reply:
This is declared const in https://github.com/mixxxdj/mixxx/blob/8790b5fbf2eb555da740a16ae39cf08215868031/src/engine/sync/syncable.h#L92 I can not change this modifying other parts of the sync code. And I want not expand the scope of the feature PR.

Reply:
Reading it again, I was wrong, this was just the method, not the member.